test/Array/array_copyWithin.js

```diff
@@ -0,0 +1,17 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Check for regressions on an issue where copyWithin was failing to look up properties from the prototype.
+var a21_proto = {};
+Object.defineProperty(a21_proto, "0", {

```

Comment:
could you add tests where proto is an proxy object and you can see has/get called properly.

Reply:
Good idea, thanks!